azure-pipelines.yml

```diff
@@ -330,16 +330,17 @@ stages:
         artifactName: Source_Build_Logs
         artifactType: Container
         parallel: true
-    - task: PublishBuildArtifacts@1
-      displayName: Upload package artifacts
-      # Only capture source build artifacts in PRs for the sake of inspecting
-      # changes that impact source-build. The artifacts from this build pipeline are never actually used.
-      condition: and(succeeded(), in(variables['Build.Reason'], 'PullRequest'))
-      inputs:
-        pathtoPublish: artifacts/packages/
-        artifactName: Source_Build_Packages
-        artifactType: Container
-        parallel: true
+    - ${{ if eq(variables['_extensionsBuildProducedPackages'], true) }}:
```

Comment:
This won't work as this is evaluated at compile time and the variable will be defined at runtime, so this will evaluate to: 

`eq(Null, true)` and this step will never be executed. You need to use a runtime condition. 

```
condition: and(succeeded(), in(variables['Build.Reason'], 'PullRequest', eq(variables['_extensionsBuildProducedPackages'], true)))
```

Reply:
```
Evaluating: and(succeeded(), in(variables['Build.Reason'], 'PullRequest'), eq(variables['_extensionsBuildProducedPackages'], True))
Expanded: and(True, in('PullRequest', 'PullRequest'), eq(Null, True))
Result: False
```

its still getting evaluated to Null

Reply:
Yeah, which is what we want, for this to run only when the variable is set. However in the other case, even if the variable was set, it will always evaluate to `Null`.